### PR TITLE
Enable fallback synthetic source for `point` and `shape`

### PR DESCRIPTION
**Pull request by lkts**
This PR enables fallback synthetic source for `point` and `shape` fields and adds appropriate tests.

Contributes to #106460.

### CODE REVIEW

File: x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java [hunk 1]
```diff
@@ -419,19 +431,169 @@ protected Object generateRandomInputValue(MappedFieldType ft) {
...
+                    // When reading doc-values, the block is a list of encoded longs
+                    List<Long> outBlockList = values.stream()
+                        .map(Value::point)
+                        .filter(Objects::nonNull)
+                        .map(this::encode)
+                        .sorted()
+                        .toList();
+                    Object outBlock = outBlockList.size() == 1 ? outBlockList.get(0) : outBlockList;
+                    return new SyntheticSourceExample(representations, representations, outBlock, this::mapping);
+                } else {
```
Comment (kkrik-es) [inline x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java:473]: Nit: skip else after return.

File: x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java [hunk 2]
```diff
@@ -419,19 +431,169 @@ protected Object generateRandomInputValue(MappedFieldType ft) {
...
+                    // When reading row-stride, the block is a list of WKT encoded BytesRefs
+                    List<String> outBlockList = values.stream()
+                        .map(Value::point)
+                        .filter(Objects::nonNull)
+                        .map(CartesianPoint::toWKT)
+                        .toList();
+                    Object outBlock = outBlockList.size() == 1 ? outBlockList.get(0) : outBlockList;
```
Comment (kkrik-es) [inline x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java:480]: Can the list be empty, e.g. if all values are null?

**Review by kkrik-es — APPROVED** [review]

**Review by martijnvg — APPROVED** [review]
👍

**Comment by lkts** [comment]
@elasticmachine run elasticsearch-ci/7.17.22 / bwc-snapshots

**Comment by lkts** [comment]
buildkite test this

File: x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java (hunk 2 above)
Comment (lkts) [reply x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapperTests.java:480]: Yes but it is handled in test code separately.

**Pull request merged**